src/main/java/com/clouway/push/client/PushChannelApiImpl.java

```diff
@@ -239,4 +248,10 @@ public void onTokenExpire() {
       }
     });
   }
+
+  private static native void notifyForNewMessage(JavaScriptObject o) /*-{
+    if ($wnd.onChannelMessageReceived) {
+      $wnd.onChannelMessageReceived(o);
```

Comment:
I think this part will not be necessary. I've almost implemented the angular side of the functionality.